### PR TITLE
Support Pinot with secure connection

### PR DESCRIPTION
**Pull request by xiangfu0**
## Description

Support pinot TLS connection to controller/broker when the `pinot.controller-urls` configuration uses `https` as the scheme.

## Release notes

```markdown
# Pinot
Support TLS connection when the `pinot.controller-urls` configuration uses `https` as the scheme.
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/pinot.rst [hunk 1]
```diff
@@ -74,6 +74,7 @@ Property name                                             Required   Description
 ``pinot.aggregation-pushdown.enabled``                    No         Push down aggregation queries, default is ``true``.
 ``pinot.count-distinct-pushdown.enabled``                 No         Push down count distinct queries to Pinot, default is ``true``.
 ``pinot.target-segment-page-size``                        No         Max allowed page size for segment query, default is ``1MB``.
+``pinot.secure-connection``                               No         Use https for all connections, default is ``false``.
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/pinot.rst:77]: Please change to `pinot.tls.enabled` for consistency with other connectors.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 2]
```diff
@@ -238,6 +239,18 @@ public PinotConfig setGrpcEnabled(boolean grpcEnabled)
         return this;
     }
 
+    public boolean isSecureConnection()
+    {
+        return secureConnection;
+    }
+
+    @Config("pinot.secure-connection")
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:247]: Please add an assertion to verify no conflict with `pinot.controller-urls`. For instance, when this config property is true, `pinot.controller-urls` shouldn't have `https` url.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 3]
```diff
@@ -115,11 +119,12 @@
     private static final String TABLE_SCHEMA_API_TEMPLATE = "tables/%s/schema";
     private static final String ROUTING_TABLE_API_TEMPLATE = "debug/routingTable/%s";
     private static final String TIME_BOUNDARY_API_TEMPLATE = "debug/timeBoundary/%s";
-    private static final String QUERY_URL_TEMPLATE = "http://%s/query/sql";
+    private static final String QUERY_URL_TEMPLATE = "query/sql";
 
     private final List<URI> controllerUrls;
     private final HttpClient httpClient;
     private final PinotHostMapper pinotHostMapper;
+    private final boolean secureConnection;
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:127]: Please define `String scheme` instead and remove `secureConnection ? HTTPS_SCHEME : HTTP_SCHEME`.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 4]
```diff
@@ -217,8 +223,12 @@ private <T> T sendHttpGetToControllerJson(String path, JsonCodec<T> codec)
     {
         ImmutableMultimap.Builder<String, String> additionalHeadersBuilder = ImmutableMultimap.builder();
         controllerAuthenticationProvider.getAuthenticationToken().ifPresent(token -> additionalHeadersBuilder.put(AUTHORIZATION, token));
+        final URI controllerPathUri = uriBuilderFrom(getControllerUrl())
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:226]: nit: Remove local `final`.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 5]
```diff
@@ -228,8 +238,13 @@ private <T> T sendHttpGetToBrokerJson(String table, String path, JsonCodec<T> co
     {
         ImmutableMultimap.Builder<String, String> additionalHeadersBuilder = ImmutableMultimap.builder();
         brokerAuthenticationProvider.getAuthenticationToken().ifPresent(token -> additionalHeadersBuilder.put(AUTHORIZATION, token));
+        final HttpUriBuilder builder = HttpUriBuilder.uriBuilder().hostAndPort(HostAndPort.fromString(getBrokerHost(table)));
+        final URI brokerPathUri = builder
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:242]: nit: Remove local `final`.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 6]
```diff
@@ -530,10 +545,14 @@ private BrokerResponseNative submitBrokerQueryJson(ConnectorSession session, Pin
     {
         String queryRequest = QUERY_REQUEST_JSON_CODEC.toJson(new QueryRequest(query.getQuery()));
         return doWithRetries(PinotSessionProperties.getPinotRetryCount(session), retryNumber -> {
-            String queryHost = getBrokerHost(query.getTable());
-            LOG.info("Query '%s' on broker host '%s'", queryHost, query.getQuery());
-            Request.Builder builder = Request.Builder.preparePost()
-                    .setUri(URI.create(format(QUERY_URL_TEMPLATE, queryHost)));
+            final HttpUriBuilder httpUriBuilder = HttpUriBuilder.uriBuilder().hostAndPort(HostAndPort.fromString(getBrokerHost(query.getTable())));
+            final URI queryPathUri = httpUriBuilder
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:549]: nit: Remove local `final`.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java [hunk 7]
```diff
@@ -193,17 +193,25 @@ public TlsGrpcQueryClientFactory(PinotGrpcServerQueryClientConfig grpcClientConf
...
-                    .buildOrThrow());
+                    .put(CONFIG_USE_PLAIN_TEXT, String.valueOf(grpcClientConfig.isUsePlainText()));
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_TYPE, tlsConfig.getKeystoreType());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_PATH, tlsConfig.getKeystorePath());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_PASSWORD, tlsConfig.getKeystorePassword());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_TYPE, tlsConfig.getTruststoreType());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_PATH, tlsConfig.getTruststorePath());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_PASSWORD, tlsConfig.getTruststorePassword());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + SSL_PROVIDER, tlsConfig.getSslProvider());
+            this.config = new GrpcQueryClient.Config(tlsConfigBuilder.buildOrThrow());
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java:206]: Could you explain why these change are needed in this PR?

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java [hunk 8]
```diff
@@ -243,9 +251,10 @@ public Iterator<PinotDataTableWithSize> queryPinot(ConnectorSession session, Str
             });
             BrokerRequest brokerRequest = REQUEST_COMPILER.compileToBrokerRequest(query);
             GrpcRequestBuilder requestBuilder = new GrpcRequestBuilder()
-                    .setSql(query)
                     .setSegments(segments)
                     .setEnableStreaming(true)
+                    .setBrokerId("trino-coordinator-grpc")
+                    .setSql(query)
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java:257]: Please revert the re-ordering of `setSql()`. It unrelates to https change.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java (hunk 7 above)
Comment (xiangfu0) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java:206]: The issue is Pinot Config does not like null as the value, and throws exceptions when initializing grpc client, hence, we need to ignore the property instead of setting it as null.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 9]
```diff
@@ -238,6 +239,26 @@ public PinotConfig setGrpcEnabled(boolean grpcEnabled)
...
+    public boolean isTlsEnabled()
+    {
+        return tlsEnabled;
+    }
+
+    @Config("pinot.tls.enabled")
+    public PinotConfig setTlsEnabled(boolean tlsEnabled)
+    {
+        this.tlsEnabled = tlsEnabled;
+        if (tlsEnabled && (controllerUrls != null)) {
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:251]: Please extract a method. You can refer to `MySqlJdbcConfig` class.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 10]
```diff
@@ -104,6 +106,8 @@
 {
     private static final Logger LOG = Logger.get(PinotClient.class);
     private static final String APPLICATION_JSON = "application/json";
+    private static final String HTTPS_SCHEME = "https";
+    private static final String HTTP_SCHEME = "http";
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:110]: nit: Inline.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java (hunk 7 above)
Comment (ebyhr) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java:206]: I know it. I'm asking "why these change are needed in this PR". Do you mean this change relates to https commit?

Comment (xiangfu0) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java:206]: The reason is when enable TLS in Pinot for HTTPS and grpc together. The initialization for the pinot grpc client will fail. So I cannot perform the whole e2e testing.

It's also possible to fix this as a separated PR, but I don't feel like it's worth it, cause typical users won't enable tls for grpc but leave controller/broker non-tls.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java (hunk 9 above)
Comment (xiangfu0) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java]: 👍

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 11]
```diff
@@ -257,4 +271,19 @@ public void validate()
...
+    @AssertTrue(message = "Controller URL should use HTTPS scheme when TLS is enabled for Pinot connector")
+    public boolean isControllerUrlValid()
+    {
+        if (!isTlsEnabled()) {
+            return true;
+        }
+        for (URI url : getControllerUrls()) {
+            String scheme = url.getScheme();
+            if (scheme != null && !"https".equalsIgnoreCase(scheme)) {
+                return false;
```
Comment (ebyhr) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:284]: When a scheme is omitted in `pinot.controller-urls` & `pinot.tls.enabled` is enabled, users will expect using https. This should be fixed or improve the error message (or document).

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java (hunk 7 above)
Comment (ebyhr) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java:206]: Thanks, I get it now. First of all, we should use `Optional` type in the config properties instead of nullable `String`. Please left a TODO comment.

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java [hunk 12]
```diff
@@ -97,4 +102,20 @@ public void testInvalidCountDistinctPushdown()
                 .isInstanceOf(IllegalStateException.class)
                 .hasMessage("Invalid configuration: pinot.aggregation-pushdown.enabled must be enabled if pinot.count-distinct-pushdown.enabled");
     }
+
+    @Test
+    public void testControllerUrls()
+    {
+        assertTrue(isControllerUrlValid("http://my-controller-1:9000,http://my-controller-2:9000", false));
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java:109]: Let's add more cases:
* https, false
* no scheme, true
* no scheme, false

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java (hunk 11 above)
Comment (xiangfu0) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:284]: Right, the error msg will show that when tls is enabled, the scheme should be https.

Comment (xiangfu0) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:299]: I separated the check to two for tls enabled and tls disabled, so it will give more explicit message for users.

**Review by ddcprg — COMMENTED** [review]
overall looks good to me with the requested changes, I'll let @ebyhr approve

**Comment by xiangfu0** [comment]
@ebyhr  can you take another pass for this

**Review by elonazoulay — APPROVED** [review]
Overall lgtm!

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java [hunk 13]
```diff
@@ -97,4 +102,23 @@ public void testInvalidCountDistinctPushdown()
...
+        assertTrue(isControllerUrlValid("https://my-controller-1:8443,https://my-controller-2:8443", true));
+        assertFalse(isControllerUrlValid("http://my-controller-1:9000,http://my-controller-2:9000", true));
+        assertFalse(isControllerUrlValid("https://my-controller-1:8443,https://my-controller-2:8443", false));
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java]: true. I will not check `isControllerUrlValidWhenTlsDisabled` for now then, so the default behavior is not impacted.

[the remark this answers is not in the document]
Comment (ebyhr) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java]: Just removing `isControllerUrlValidWhenTlsDisabled` doesn't make sense to me. What do you think about following?  

> I think we should control pinot.controller-urls by this flag only when users omitted scheme

[the remark this answers is not in the document]
Comment (xiangfu0) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java]: Current `pinot.controller-urls` directly parse string to URI, so it gives default scheme HTTP if empty.
If we want to change it, there are two ways:
1. For backward compatibility:
Since user only set `pinot.controller-urls`, whenever we detect HTTPS, we override the `tlsEnable` to `true`.

2. When users set both fields:
2.1. If `pinot.controller-urls` is set first and it's HTTPS, then  `pinot.tls.enabled` is already true. And user should still set`pinot.tls.enabled=true`, otherwise the validation will raise an exception.
2.2. If `pinot.tls.enabled=true` is set first, then if `pinot.controller-urls` is set HTTPS, it works, otherwise, we can still throw exception.

So the change can be:
In controller-uri set method, try to get the scheme and only set `tlsEnable` to `true` when the scheme is HTTPS.

[the remark this answers is not in the document]
Comment (xiangfu0) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConfig.java]: Changed tlsEnable from boolean to Boolean to simplify the code.

**Comment by xiangfu0** [comment]
@ebyhr can you take another pass?

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 14]
```diff
@@ -258,4 +272,38 @@ public void validate()
                 !countDistinctPushdownEnabled || aggregationPushdownEnabled,
                 "Invalid configuration: pinot.aggregation-pushdown.enabled must be enabled if pinot.count-distinct-pushdown.enabled");
     }
+
+    @AssertTrue(message = "Controller URL should use HTTPS scheme when TLS is enabled")
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:276]: Why not just decide whether to use TLS or not based on the url scheme. It seems better than having two flags that need to agree.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 15]
```diff
@@ -258,4 +265,19 @@ public void validate()
                 !countDistinctPushdownEnabled || aggregationPushdownEnabled,
                 "Invalid configuration: pinot.aggregation-pushdown.enabled must be enabled if pinot.count-distinct-pushdown.enabled");
     }
+
+    @AssertTrue(message = "All controller URLs have some scheme")
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:269]: "All controller URLs must have *the same* scheme"

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 16]
```diff
@@ -258,4 +265,19 @@ public void validate()
...
+        String scheme = null;
+        for (URI uri : getControllerUrls()) {
+            if (scheme == null) {
+                scheme = uri.getScheme();
+            }
+            else if (!StringUtils.equals(scheme, uri.getScheme())) {
+                return false;
+            }
+        }
+        return true;
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:281]:
```
return controllerUrls.stream()
        .map(URI::getScheme)
        .distinct()
        .count() == 1;
```

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 17]
```diff
@@ -258,4 +265,19 @@ public void validate()
                 !countDistinctPushdownEnabled || aggregationPushdownEnabled,
                 "Invalid configuration: pinot.aggregation-pushdown.enabled must be enabled if pinot.count-distinct-pushdown.enabled");
     }
+
+    @AssertTrue(message = "All controller URLs have some scheme")
+    public boolean validateControllerUrlsScheme()
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:270]: This method doesn't do any validation. It just returns whether all schemes are the same. So, rename it to `allUrlSchemesEqual`

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java (hunk 16 above)
Comment (xiangfu0) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java]: I should really learn new java syntax !

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java [hunk 18]
```diff
@@ -239,6 +240,11 @@ public PinotConfig setGrpcEnabled(boolean grpcEnabled)
         return this;
     }
 
+    public Boolean isTlsEnabled()
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotConfig.java:243]: Return `boolean`. This method can (and should) never return null.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 19]
```diff
@@ -152,6 +155,7 @@ public PinotClient(
         this.brokerResponseCodec = requireNonNull(brokerResponseCodec, "brokerResponseCodec is null");
         requireNonNull(config, "config is null");
         this.pinotHostMapper = requireNonNull(pinotHostMapper, "pinotHostMapper is null");
+        this.scheme = (config.isTlsEnabled() == null) ? null : (config.isTlsEnabled() ? "https" : "http");
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:158]: `config.isTlsEnabled()` should never return null, so this should be:

```java
this.scheme = config.isTlsEnabled() ? "https" : "http";
```

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 20]
```diff
@@ -213,8 +217,12 @@ private <T> T sendHttpGetToControllerJson(String path, JsonCodec<T> codec)
     {
         ImmutableMultimap.Builder<String, String> additionalHeadersBuilder = ImmutableMultimap.builder();
         controllerAuthenticationProvider.getAuthenticationToken().ifPresent(token -> additionalHeadersBuilder.put(AUTHORIZATION, token));
+        HttpUriBuilder controllerPathUriBuilder = uriBuilderFrom(getControllerUrl()).appendPath(path);
+        if (scheme != null) {
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:221]: `scheme` should never be null

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 21]
```diff
@@ -224,8 +232,16 @@ private <T> T sendHttpGetToBrokerJson(String table, String path, JsonCodec<T> co
     {
         ImmutableMultimap.Builder<String, String> additionalHeadersBuilder = ImmutableMultimap.builder();
         brokerAuthenticationProvider.getAuthenticationToken().ifPresent(token -> additionalHeadersBuilder.put(AUTHORIZATION, token));
+        HttpUriBuilder builder = HttpUriBuilder.uriBuilder().hostAndPort(HostAndPort.fromString(getBrokerHost(table)));
+        if (scheme != null) {
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:236]: `scheme` should never be null

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 22]
```diff
@@ -526,10 +542,19 @@ private BrokerResponseNative submitBrokerQueryJson(ConnectorSession session, Pin
     {
         String queryRequest = QUERY_REQUEST_JSON_CODEC.toJson(new QueryRequest(query.getQuery()));
         return doWithRetries(PinotSessionProperties.getPinotRetryCount(session), retryNumber -> {
-            String queryHost = getBrokerHost(query.getTable());
-            LOG.info("Query '%s' on broker host '%s'", queryHost, query.getQuery());
-            Request.Builder builder = Request.Builder.preparePost()
-                    .setUri(URI.create(format(QUERY_URL_TEMPLATE, queryHost)));
+            HttpUriBuilder httpUriBuilder = HttpUriBuilder.uriBuilder().hostAndPort(HostAndPort.fromString(getBrokerHost(query.getTable())));
+            if (scheme != null) {
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:546]: `scheme` should never be null

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 23]
```diff
@@ -114,11 +116,12 @@
     private static final String TABLE_SCHEMA_API_TEMPLATE = "tables/%s/schema";
     private static final String ROUTING_TABLE_API_TEMPLATE = "debug/routingTable/%s";
     private static final String TIME_BOUNDARY_API_TEMPLATE = "debug/timeBoundary/%s";
-    private static final String QUERY_URL_TEMPLATE = "http://%s/query/sql";
+    private static final String QUERY_URL_TEMPLATE = "query/sql";
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:119]: This is no longer a "template". Rename the constant to, for instance, `QUERY_URL_PATH`

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java [hunk 24]
```diff
@@ -250,6 +250,7 @@ public Iterator<PinotDataTableWithSize> queryPinot(ConnectorSession session, Str
                     .setSql(query)
                     .setSegments(segments)
                     .setEnableStreaming(true)
+                    .setBrokerId("trino-coordinator-grpc")
```
Comment (martint) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java:253]: How is this related?

Comment (xiangfu0) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java]: removed

**Comment by xiangfu0** [comment]
Thanks @martint !
